lib/auth/keystore/aws_kms.go

```diff
@@ -388,12 +401,13 @@ func (a *awsKMSKeystore) deleteUnusedKeys(ctx context.Context, activeKeys [][]by
 			return nil
 		}
 
-		clusterName := a.clusterName.GetClusterName()
-		if !slices.ContainsFunc(output.Tags, func(tag kmstypes.Tag) bool {
-			return aws.ToString(tag.TagKey) == clusterTagKey && aws.ToString(tag.TagValue) == clusterName
-		}) {
-			// This key was not created by this Teleport cluster, never delete it.
-			return nil
+		// All tags must match for this key to be considered for deletion.
+		for k, v := range a.tags {
+			if !slices.ContainsFunc(output.Tags, func(tag kmstypes.Tag) bool {
+				return aws.ToString(tag.TagKey) == k && aws.ToString(tag.TagValue) == v
+			}) {
+				return nil
+			}
 		}
```

Comment:
Let me think about this some more. Perhaps the best path is to keep the TeleportCluster tag have the tags field allow for additional tags.

Reply:
What do you think of [730d1dc](https://github.com/gravitational/teleport/pull/48132/commits/730d1dc2d1774a40da4640499548e7dffb17018e)

It will add the `TeleportCluster` tag if its not specified in the tag configuration.